test/types/string.js

```diff
@@ -561,7 +551,8 @@ describe('string', () => {
             Helper.validate(schema, [
                 ['joe@example.com', true],
                 ['joe@www.example.com', true],
-                ['joe@localhost', false, null, '"value" must be a valid email'],
```

Comment:
Makes me wonder if this should be a breaking change for joi or if it's considered a bugfix from your pov.